PurchasesCoreSwift/Misc/SystemInfo.swift

```diff
@@ -0,0 +1,169 @@
+        return Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? ""
+    }
+
+    @objc public static var platformHeader: String {
+        return Self.forceUniversalAppStore ? "iOS" : platformHeaderConstant
+    }
+
+    @objc public static var identifierForVendor: String? {
+        #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+            return UIDevice.current.identifierForVendor?.uuidString
+        #elseif os(watchOS)
+            return WKInterfaceDevice.current().identifierForVendor?.uuidString
+        #else
+            return nil
+        #endif
+    }
+
+    private static var privateProxyURL: URL?
+    @objc public static var proxyURL: URL? {
+        get {
+            return privateProxyURL
+        }
+        set {
+            privateProxyURL = newValue
+            if let privateProxyURLString = newValue?.absoluteString {
+                Logger.info(Strings.configure.configuring_purchases_proxy_url_set
+                                .replacingOccurrences(of: "%@", with: privateProxyURLString))
+            }
+        }
+    }
+
+    static var defaultServerHostURL: URL {
+        return URL(string: defaultServerHostName)!
+    }
+
+    @objc public static var serverHostURL: URL {
+        return self.proxyURL ?? Self.defaultServerHostURL
+    }
+
+    private var isApplicationBackgrounded: Bool {
+    #if os(iOS) || (targetEnvironment(simulator) && os(iOS))
+        return self.isApplicationBackgroundedIOS
+    #elseif os(tvOS)
+        return UIApplication.shared.applicationState == UIApplication.State.background
+    #elseif os(macOS)
+        return false
+    #elseif os(watchOS)
+        return  WKExtension.shared().applicationState == WKApplicationState.background
+    #endif
+    }
+
+    #if os(iOS)
+    // iOS App extensions can't access UIApplication.sharedApplication, and will fail to compile if any calls to
+    // it are made. There are no pre-processor macros available to check if the code is running in an app extension,
+    // so we check if we're running in an app extension at runtime, and if not, we use KVC to call sharedApplication.
+    @objc private var isApplicationBackgroundedIOS: Bool {
+        if self.isAppExtension {
+            return true
+        }
+
+        return UIApplication.shared.applicationState == UIApplication.State.background
+    }
+
+    @objc private var isAppExtension: Bool {
+        return Bundle.main.bundlePath.hasSuffix(".appex")
+    }
+    #endif
+
+    @objc required public init(platformFlavor: String?, platformFlavorVersion: String?, finishTransactions: Bool) throws {
+        if let platformFlavor = platformFlavor {
+            self.platformFlavor = platformFlavor
+        } else {
+            self.platformFlavor = "native"
+        }
+
+        if let platformFlavorVersion = platformFlavorVersion {
+            self.platformFlavorVersion = platformFlavorVersion
+        } else {
+            self.platformFlavorVersion = nil
```

Comment:
this if / else isn't really necessary, right? i.e.: just assigning to the optional would be the same?

Reply:
Definitely, yeah. This was part of my debugging. I was trying to make sure I wasn't somehow missing something so I made if/else super explicit for a minute. Then forgot to go back and revise.
🤦‍♂️ again.